client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -419,8 +423,8 @@ public void write(PendingEvent event) {
                 // if connection is null getConnection() establishes a connection and retransmits all events in inflight
                 // list.
                 connection = Futures.getThrowingException(getConnection());
-            } catch (SegmentSealedException e) {
-                // Add the event to inflight and indicate to the caller that the segment is sealed.
+            } catch (SegmentSealedException | NoSuchSegmentException e) {
```

Comment:
I'm confused about this change. This line change is in the write call, and the description of the issue refers to getting no such segment while setting up append.

Reply:
On receiving a `NoSuchSegment`  the response processor callback invokes  `state.failConnection(new NoSuchSegmentException(noSuchSegment.getSegment()))`. This will release the setupConnection exceptionally `setupConnection.releaseExceptionally( NoSuchSegmetnException)`
Now before the `resendToSuccessorsCallback` is invoked there could be writes to the same segment. When a `segment.write()` is invoked `getConnection()` method will throw a `NoSuchSegmentsException` and the event is written into the inflight queue.

All the inflight events are later resent to the successors as part of the `resendToSuccessorCallback` execution.

Reply:
Yes, the current code already adds the events to inflight incase of `NoSuchSegmentException` and `SegmentSealedException`.

> One general question: is there any exception that we can get here with which we would not want to add to in-flight?

Any connection failure exception/ Auth related exception/ processingFailure exception will be thrown here ( a reconnect will be triggered in the background.)